src/clippoly/src/clippoly.cpp

```diff
@@ -0,0 +1,295 @@
+// Copyright 2021 The Manifold Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "clippoly.h"
+
+#include <clipper2/clipper.h>
+
+#include <vector>
+
+#include "clipper2/clipper.core.h"
+#include "clipper2/clipper.engine.h"
+#include "clipper2/clipper.offset.h"
+#include "glm/ext/vector_float2.hpp"
+#include "public.h"
+
+using namespace manifold;
+
+namespace {
+const int precision_ = 8;
```

Comment:
Is this absolute or relative?

Reply:
Number of decimal places (they use int64 internally, so this controls the conversion)

Reply:
Okay, since this is absolute precision like our own `Precision` in Manifold, we should use a similar bounding box computation to determine the right value. Otherwise if someone starts using meters and gets tiny numbers, everything will just collapse to a point.

Reply:
I should also mention that 8 is the maximum value.

Reply:
Oh, interesting. Any idea why that is? Do they expect us to scale things into a range closer to 1 before input?

Reply:
No sorry, I haven't actually gone digging about why it stops at 8 with 10 digits of headroom in int64 😅. Clipper used to not even have the double interface at all. If more flexibility than 8 is required, we'd need to use the int64 side and do the precision scaling in CrossSection.

Reply:
Okay, good enough for now, but maybe file an issue so we don't forget when this gets merged.